src/approaches/grammar_search_invention_approach.py

```diff
@@ -236,16 +350,29 @@ def enumerate(self) -> Iterator[Tuple[Predicate, float]]:
             yield (predicate, cost)
             classifier = _ForallClassifier(predicate)
             yield (Predicate(str(classifier), [], classifier), cost)
+            if predicate.arity >= 2:
```

Comment:
When will this ever happen, since we're only considering UnaryPredicates right now?

Reply:
The given predicates can have arity 2 (e.g. `On`)..